python/tvm/driver/build_module.py

```diff
@@ -181,7 +181,7 @@ def lower(sch,
         tvm.tir.transform.BF16Legalize(),
         tvm.tir.transform.NarrowDataType(32),
         tvm.tir.transform.Simplify(),
-        tvm.tir.transform.HoistIfThenElse(),
+        tvm.tir.transform.HoistIfThenElse("basic"),
```

Comment:
Do we really need this "basic" pass? Can we directly perform a complete pass in Phase 3?

Reply:
Yes, we can combine both also and place the pass in End phase. 
I understand the dilemma here. Even i also had the same :)
The reasoning i came up later is as below.
Current PR supports the feature for "Block scope vars" or "Attr nodes" which happens to be more applicable in specific cases(For example in Cuda Kernels). Also there is slight increase in time complexity(As linear).

So to sum up, we have 2 cases : 
Case 1: "Basic" or "Default": The scenarios covered here should be more general(simpler version) across.
Case 2: "Advanced" : The scenarios covered here should be enabled in case of particular settings.

Please let me know your thought on above.

Reply:
I don't think a little more compile-time complexity is a real problem. Perhaps the pass can even be faster than those Python bindings. We can always perform the "advanced" version as long as it won't break the correctness or do negative optimizations.

Reply:
Yes, you are right! The compile-time complexity is not the real point here. I just shared the info :)
I am okay for the both the approach. 

However i think it would be good to have one additional segregation inside Pass to have more control on the different scenarios it has covered.  Which can provide more user friendly experience, when user wants to club the Pass with only specific Passes without needing to write special config parameters.

Let us have some more opinion on this point, to conclude better.
@MarisaKirisame : Would please help share your thoughts on above point. TIA!

Reply:
I think we should use the advanced option. It is more general and the compile time is light juding from the code.

Reply:
Thanks @MarisaKirisame for sharing your thoughts here.
With this i think we can conclude the comment here, by removing the "Basic" pass embedding in the lower sequence of passes, and place only one hoisting pass at the third phase as per latest change.

If my understanding is wrong, please let me know. Thanks!

Reply:
That is good for me.